internal/controller/bucket/bucket.go

```diff
@@ -417,6 +383,20 @@ WAIT:
 			c.log.Info("Context timeout", "bucket_name", bucket.Name)
 
 			return managed.ExternalCreation{}, ctx.Err()
+		case beName := <-readyChan:
+			c.log.Info("Bucket created", "backend name", beName, "bucket_name", bucket.Name)
+
+			bucket.Status.SetConditions(xpv1.Available())
+			bucket.Status.AtProvider.BackendStatuses = v1alpha1.BackendStatuses{
```

Comment:
There is no existing statuses, because create only triggered if it is empty. Other backend calls don't update the object. So once this code inserts the first item, it goes to update loop, which will fulfill the object with all backend statuses. So we can use the helper, but the code works as design.

Reply:
Ah yes, makes sense :+1: